### PR TITLE
Construct database models

### PR DESCRIPTION
**Pull request by yuna878**
## Overview

Construct database models as a first step to implement centralized data storage.

## Changes Made

- Create connection to SQLite database
- Construct database models
  - These models are inconsistent with GraphQL schema because we want to store data in a logical manner rather than mirroring the data structure that Cornell Dining gives. This means we need to refactor current parsing methods to : 1) parse Cornell Dining data and store to db 2) query desired data from db and format to match GraphQL schema.
  - All the models extend `Base` class so that tables are able to reference other tables with relationship.

## Test Coverage

Wrote up a short script to test that tables can be generated easily with models and that schemas are created correctly

Script:
Screen Shot 2019-10-23 at 2 45 04 AM

Result:
Screen Shot 2019-10-23 at 2 45 31 AM

## Next Steps

Future PRs should address the refactoring of parsing functions as mentioned above. When transition to using a database is done, we will have to write up a cron job to run the updates to database.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -24,3 +24,6 @@ venv/
 
 # Swipe Data
 eatery-data/
+
+# DB
+data.db
```
Comment (young-k) [inline .gitignore]: I'm pretty certain that the naming convention for sqlite files is `.sqlite3`

File: src/db_models/CampusHoursBase.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    def __tablename__(cls):
+        class_name = cls.__name__  # e.g. TrilliumHour
+        table_name = class_name[0].lower() + class_name[1:] + "s"  # e.g. trilliumHours
+        return table_name
+
+    date = Column(String, nullable=False)
+    event_description = Column(String, nullable=False)
+    event_summary = Column(String, nullable=False)
+    end_time = Column(String, nullable=False)
+    id = Column(Integer, nullable=False, primary_key=True)
```
Comment (young-k) [inline src/db_models/CampusHoursBase.py]: It's normally convention to have `id` always be the first field of any model, but it's up to @meganle

File: src/db_models/CollegetownHoursBase.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+from sqlalchemy import Column, Integer, String
+from sqlalchemy.ext.declarative import declared_attr
+
+
+class CollegetownHoursBase(object):
+    @declared_attr
+    def __tablename__(cls):
+        class_name = cls.__name__  # e.g. PokelavaHour
+        table_name = class_name[0].lower() + class_name[1:] + "s"  # e.g. pokelavaHours
+        return table_name
```
Comment (young-k) [inline src/db_models/CollegetownHoursBase.py]: Could there be a base class that defines this `tablename` function that `CampusHoursBase` and `CollegetownHoursBase` both extend from?

File: src/db_models/MenuBase.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+from sqlalchemy import Boolean, Column, ForeignKey, Integer, String
+from sqlalchemy.ext.declarative import declared_attr
+
+
+class MenuBase(object):
+    @declared_attr
+    def __tablename__(cls):
+        class_name = cls.__name__  # e.g. TrilliumMenu
+        table_name = class_name[0].lower() + class_name[1:] + "s"  # e.g. trilliumMenus
+        return table_name
```
Comment (young-k) [inline src/db_models/MenuBase.py]: same comment here

File: src/db_models/SwipeDataBase.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+from sqlalchemy import Column, Float, Integer, String
+from sqlalchemy.ext.declarative import declared_attr
+
+
+class SwipeDataBase(object):
+    @declared_attr
+    def __tablename__(cls):
+        class_name = cls.__name__  # e.g. TrilliumSwipeData
+        table_name = class_name[0].lower() + class_name[1:] + "s"  # e.g. trilliumSwipeDatas
+        return table_name
```
Comment (young-k) [inline src/db_models/SwipeDataBase.py]: and again here!

**Review by young-k — CHANGES_REQUESTED** [review]
Minor changes, but looks good!

**Review by meganle — APPROVED** [review]
Looks good!! 🔥

**Review by orkosinha — APPROVED** [review]
lgtm!!

File: requirements.txt [hunk 6]
```diff
@@ -9,6 +9,7 @@ file-read-backwards==2.0.0
 Flask==1.0.2
 Flask-GraphQL==2.0.0
 graphene==2.1.3
+graphene-sqlalchemy==2.2.2
```
Comment (meganle) [inline requirements.txt]: do we need?

**Review by young-k — APPROVED** [review]
lgtm

**Pull request merged**